chain/index/chain-indexing-overview-for-rpc-providers.MD

```diff
@@ -60,34 +71,137 @@ The ChainIndexer [periodically runs](https://github.com/filecoin-project/lotus/b
 
-1. **Stop the Lotus Node**
-   - Stop your Lotus node before starting the upgrade and migration process.
-2. **Update Configuration**
-   - Modify your Lotus configuration to enable the `ChainIndexer` as described in the [`ChainIndexer Config` section above](#chainindexer-config).
-3. **Restart Lotus Node**
-   - Restart your Lotus node with the new configuration.
-   - The `ChainIndexer` will begin indexing **real-time chain state changes** immediately in the `${LOTUS_PATH}/chainindex` directory.
-   - **However, it will not automatically index any historical chain state (i.e., any previously existing chain state prior to the upgrade). To perform backfilling, please see the [`Backfill` section below](#backfill).**
+### Backfill Disk Space Requirements
 
-> **Note:** It's recommended to keep the [pre-ChainIndexer](#previous-indexing-system) indexing database directory (`${LOTUS_PATH}/sqlite`) around until you've confirmed you don't need to [downgrade](#downgrade).  After sustained successful operations after the upgrade, the [pre-ChainIndexer](#previous-indexing-system) indexing database directory can be removed to reclaim disk space.  
+As of 202410, ChainIndexer will accumulate approximately ~340 MiB per day of data, or 10 GiB per month (see [here](https://github.com/filecoin-project/lotus/issues/12453)).
 
-## Backfill
-There is no automated migration from [pre-ChainIndexer indices](#previous-indexing-system) to the [ChainIndex](#chainindexer-indexing-system).  Instead one needs to index historical chain state (i.e., backfill) using one of the following tools.
+### `lotus-shed chainindex validate-backfill` CLI tool
+The `lotus-shed chainindex validate-backfill` command is a tool for validating and optionally backfilling the chain index over a range of epochs since calling the [`ChainValidateIndex` API](#chainvalidateindex-rpc-api) for a single epoch at a time can be cumbersome, especially when backfilling or validating the index over a range of historical epochs, such as during a backfill. This tool wraps the `ChainValidateIndex` API to efficiently process multiple epochs.
+
+**Note: This command can only be run when the Lotus daemon is already running with the [`ChainIndexer` enabled](#enablement) as it depends on the `ChainValidateIndex` RPC API.**
+
+#### Usage
+
+```
+lotus-shed chainindex validate-backfill --from <start_epoch> --to <end_epoch> [--backfill] [--log-good]
```

Comment:
https://github.com/filecoin-project/lotus/pull/12421/files#r1802545996 I'm suggesting that `--from` be optional and it should start from head.

and thinking that though, I can see how `--epochs` would make sense as well ... but let's not overcomplicate it for now I suppose. `--from` and `--to` is nice, but head for `--epochs` is also nice.

Reply:
Per 2024-10-16 verbal we agreed not to add extra optionality and wait until this becomes a user pain point.  Resolving.